File: Frogs and Jumps.py
class Solution:
    def unvisitedLeaves(self, N, leaves, frogs):
        
        leaveStatus = [0 for j in range(leaves+1)]
        for i in range(N):
            if frogs[i]<=leaves and leaveStatus[frogs[i]]==0:
                for j in range(frogs[i], leaves+1, frogs[i]):
                    leaveStatus[j] = 1
        
        leafCount = leaves
        for i in leaveStatus:
            if(i):
                leafCount-=1
            
        
        return leafCount
    

File: Frogs and Jumps.cpp
class Solution {
  public:
    int unvisitedLeaves(int N, int leaves, int frogs[]) {
        
        vector<int> leaveStatus (leaves + 1, 0);
		for (int i = 0; i < N; i++) {
			if (frogs[i] <= leaves && leaveStatus[frogs[i]] == 0) {
				for (int j = frogs[i]; j <= leaves; j += frogs[i]) {
					leaveStatus[j] = 1;
				}
			}
		}
		int leafCount = leaves;
		for (int i : leaveStatus) {
			if (i) {
				leafCount--;
			}
		}
		return leafCount;
    }
};
